README.md

```diff
@@ -1,2 +1,191 @@
+}
+```
+
+You can then use tensorboard to visualize the scaler by using a simple command: `tensorboard --logdir .`
+
+The above snippet of code would print the time taken for the scaler to log.
+
+```
+elapsed time: 44.6872s
+```
+
+Above timing is 40s (summary creation time) + 4s (flush timing i.e time taken to write event files).
+
+Alternatively you can execute the logging in an async manner which would be faster at many instances by using `std::async`, here is a snippet which allows you to do the same operation but much faster as compared to the previous code.
+
+```
+#include <mlboard/mlboard.hpp>
+#include <iostream>
+#include <chrono> 
+#include <ctime> 
+#include <future>
+
+using namespace std;
+using namespace mlboard;
+
+// A function to mock that summary creation takes 10 sec.
+void mockfunc(const std::string& tag,
+              int step,
+              double value,
+              FileWriter& fw)
+{
+    std::this_thread::sleep_for( std::chrono::seconds(10));
+    mlboard::SummaryWriter<mlboard::FileWriter>::Scalar(tag,step,value,fw);
+}
+
+int main()
+{
+    GOOGLE_PROTOBUF_VERIFY_VERSION;
+    std::chrono::time_point<std::chrono::system_clock> start, end; 
+    start = std::chrono::system_clock::now(); 
+    FileWriter f1("temp");
+
+    std::future<void> result1 = async(std::launch::async,   mockfunc,"tag",1,1.1,std::ref(f1));
+    std::future<void> result2 = async(std::launch::async,   mockfunc, "tag",2,1.2,std::ref(f1));
+    std::future<void> result3 = async(std::launch::async,   mockfunc,"tag",3,1.3,std::ref(f1));
+    std::future<void> result4 = async(std::launch::async,  mockfunc, "tag",4,1.4,std::ref(f1));
+    result1.get();
+    result2.get();
+    result3.get();
+    result4.get();
+
+    f1.Close();  
+
+    end = std::chrono::system_clock::now(); 
+    std::chrono::duration<double> elapsed_seconds = end - start; 
+    std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
+  
+    std::cout << "finished computation at " << std::ctime(&end_time) 
+              << "elapsed time: " << elapsed_seconds.count() << "s\n"; 
+
+    google::protobuf::ShutdownProtobufLibrary();
+}
+
+```
+
+The total time of execution would be reduced to 14 sec since they all are executing parallelly.
+
+```
+elapsed time: 14.0004s
+```
+
+Above timing is 10s (summary creation time) + 4s (flush timing i.e time taken to write event files )
+
+As you can see from both the snippet, the flush timing remains same since deafult flush timing are `5000 milliseconds` and we are able to get approx same flush time, for both of the above code. The significant reduction was in summary creation time becuase of async function calling.
+
+Note: Just to benchmark, a waiting time of 10 sec was added using `std::this_thread::sleep_for( std::chrono::seconds(10));` inside the `mockfunc` (to mock a behavior of writing a summary which has a lot of data), so that there could be a clear difference between the two codes .
```

Comment:
```suggestion
Note: Just to benchmark, a waiting time of 10 sec was added using `std::this_thread::sleep_for( std::chrono::seconds(10));` inside the `mockfunc` (to mock a behavior of writing a summary which has a lot of data), so that there could be a clear difference between the two codes.
```